include/drivers/flash.h

```diff
@@ -489,6 +545,74 @@ static inline const struct flash_parameters *z_impl_flash_get_parameters(const s
 }
 #endif
 
+/**
+ * @brief Get flash page info at given offset
+ *
+ * Fills in flash_page_info structure with information on the flash page that is at given offset.
+ * In case when the dev is virtual device that begins at the middle of page the returned offset
+ * this will return error if the offset is at the split page.
+ *
+ * @param[in] dev	flash device;
+ * @param[in] off	offset;
+ * @param[out] fpi	pointer to flash_page_info for obtained information;
+ *
+ * If flash has non-uniform layout with holes, asking for offset that is within hole will return
+ * an error.
+ *
+ * @return 0 on success;
+ *	   -EINVAL if offset is outside of flash device;
+ *	   other negative errno code on error.
+ */
+__syscall int flash_get_page_info(const struct device *dev, off_t off, struct flash_page_info *fpi);
```

Comment:
copy from: https://github.com/zephyrproject-rtos/zephyr/pull/36487#discussion_r664564531
`nvlsianpu:`
isn't this similar to flash_get_page_info_by_offs()
`de-nordic:`
I have added the `flash_get_page_info`, as a part of https://github.com/zephyrproject-rtos/zephyr/issues/36485, so that it would be possible to override results returned by such call, which is not possible with the current flash_api_get_layout, because it gives direct access to objects defined within driver.
I have not removed the original calls that do layout checks, because this is proposal and I did not want to remove them yet.
The `flash_get_page_info_by_offs` and the `flash_get_page_info_by_idx`, and the `foreachs`, in case of the  https://github.com/zephyrproject-rtos/zephyr/issues/36485, would be reimplemented as non-system util library, and the CONFIG that enables them would be still available.
All these functions would be reimplemented around the new `flash_get_page_info`, `flash_get_page_count` and `flash_get_size`.
`nvlsianpu:`
Ok, understood

Reply:
This is all fine to me.